packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getListType.ts

```diff
@@ -1,14 +1,30 @@
 import { ListType } from 'roosterjs-editor-types';
 
-function isABulletList(textBeforeCursor: string) {
-    const hasTriggers = ['*', '-', '>'].indexOf(textBeforeCursor[0]) > -1;
-    const REGEX: RegExp = /^(.*?)=>|^(.*?)->|^(.*?)-->|^(.*?)=>|^(.*?)--/;
-    return hasTriggers || REGEX.test(textBeforeCursor);
+function isABulletList(trigger: string) {
+    const triggerLength = trigger.length;
+    if (triggerLength === 1) {
+        const REGEX: RegExp = /^((.*?)-|^(.*?)*|^(.*?)>)$/;
+        return REGEX.test(trigger);
+    } else if (triggerLength === 2) {
+        const REGEX: RegExp = /^((.*?)=>|^(.*?)->|^(.*?)=>|^(.*?)--)$/;
+        return REGEX.test(trigger);
+    } else if (triggerLength === 3) {
+        const REGEX: RegExp = /^(.*?)-->$/;
+        return REGEX.test(trigger);
+    }
+    return false;
 }
 
-function isANumberingList(textBeforeCursor: string) {
-    const REGEX: RegExp = /^([1-9,a-z, i,A-Z,I]{1,2}\.|[1-9,a-z, i,A-Z,I]{1,2}\)|[1-9,a-z, i,A-Z,I]{1,2}\-|\([1-9,a-z, i,A-Z,I]{1,2}\))$/;
-    return REGEX.test(textBeforeCursor.replace(/\s/g, ''));
+function isANumberingList(trigger: string) {
+    const triggerLength = trigger.length;
+    if (triggerLength === 2) {
+        const REGEX: RegExp = /^([1-9,a-z,A-Z]{1,2}\.|[1-9,a-z,A-Z]{1,2}\)|[1-9,a-z,A-Z]{1,2}\-)$/;
```

Comment:
why 1-9?  We don't support "10" ?

Reply:
And if you want to use regex, make it a global const, but no create it everytime

Reply:
And if you want to use regex, make it a global const, but no create it everytime

Reply:
And the first part seems all same, so you can do
`/^[0-9]{1,2}[\.\)\-]$/`

BTW, why we allow a-z together with 0-9? Is "1a" a valid match? Maybe what you want is
`([0-9]{1,2}|[a-z]{1,2}|[a-Z]{1,2})`

Reply:
And as I said, regex should be avoid as much as possible, you can first check if the last char is ".", ")", "-", then use regex in order to reduce the chance to exec regex.

Reply:
And as I said, regex should be avoid as much as possible, you can first check if the last char is ".", ")", "-", then use regex in order to reduce the chance to exec regex.